tests/unit/components/playback/controls/AddCustomer.spec.js

```diff
@@ -120,9 +244,274 @@ describe("AddCustomer.enter_uuidbase57", () => {
       await Vue.nextTick();
       expect(spied_text_validator).toHaveBeenCalledWith(entry);
 
-      expect(target_error_message.text()).toStrictEqual(
-        spied_text_validator.mock.results[0].value
+      expect(target_error_message.text()).toStrictEqual(response_text);
+    }
+  );
+});
+
+describe("AddCustomer.enable_save_button", () => {
+  beforeEach(async () => {
+    const propsData = {
+      dialogdata: null,
+      dataindex: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    store = await NuxtStore.createStore();
+  });
+
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  afterEach(() => wrapper.destroy());
+
+  test.each([
+    [
+      "0VSckkBYH2An3dqHEyfRRE",
+      "The entered ID has an invalid character 0,",
+      "06ad547f-fe02-477b-9473-f7977e4d5e17",
+      "",
+      "Experiment anemia -1",
+      "",
+      "color: rgb(63, 63, 63); width: 250px; left: 250px;",
+    ],
+    [
+      "5FY8KwTsQaUJ2KzHJGetfE",
+      "",
+      "06ad547f fe02-477b-9473-f7977e4d5e17",
+      "Wrong Format of API Key",
+      "Experiment anemia -1",
+      "",
+      "color: rgb(63, 63, 63); width: 250px; left: 250px;",
+    ],
+    [
+      "5FY8KwTsQaUJ2KzHJGetfE",
+      "",
+      "06ad547f-fe02-477b-9473-f7977e4d5e17",
+      "",
+      "Cat * lab",
+      "Invalid character present. Valid characters are alphanumeric & # - . _  ( ) /",
+      "color: rgb(63, 63, 63); width: 250px; left: 250px;",
+    ],
+    [
+      "5FY8KwTsQaUJ2KzHJGetfE",
+      "",
+      "06ad547f-fe02-477b-9473-f7977e4d5e17",
+      "",
+      "Experiment anemia -1",
+      "",
+      "color: rgb(255, 255, 255); width: 250px; left: 250px;",
+    ],
+  ])(
+    "Given an UUID, API Key, Nickname for 'Add Customer' as input, When the input contains based on valid the critera or failure, Then display of Label 'Save ID' is visible or greyed",
+    async (
+      uuid,
+      invalid_uuid,
+      apikey,
+      invalid_apikey,
+      nickname,
+      invalid_nickname,
+      save_btn_css
+    ) => {
+      const selector_id_suffix_alphanumeric_id = "alphanumeric-id";
+      const selector_id_suffix_apikey_id = "apikey-id";
+      const selector_id_suffix_nickname_id = "nickname-id";
+
+      const spied_text_validator_uuid = jest.spyOn(
+        TextValidation.prototype,
+        "validate_uuidBase_fiftyseven_encode"
+      );
+
+      const spied_text_validator_api = jest.spyOn(
```

Comment:
the use of these validators is already being tested elsewhere.  To try and make our tests less brittle (i.e. not have to refactor them unnecessarily), please try and only include things within a test that are necessary to test the specific "unit" being tested.

In this test, you are not checking the validation logic of individual fields----you are checking whether the Save button is enabled/disabled when certain fields are valid/invalid

Reply:
Agreed and removed.